### PR TITLE
Cleans up Cassandra Session logic in preparation of Driver v4

### PR DESCRIPTION
**Pull request by codefromthecrypt**
The more notable change in Datastax Driver v4 is consolidation of
Cluster and Session types. This centralizes logic to create a session
regardless of test or mainline. It also ensures the cluster object is
closed during main and test code when storage is.

### CODE REVIEW

File: zipkin-storage/cassandra-v1/src/test/java/zipkin2/storage/cassandra/v1/CassandraStorageExtension.java [hunk 1]
```diff
@@ -109,54 +93,27 @@ InetSocketAddress contactPoint() {
...
+        try (Session session = Access.tryToInitializeSession(contactPoint)) {
+          session.execute("SELECT now() FROM system.local");
+          logger().info("Obtained a connection to container ({})", contactPoint);
+          return null; // unused value
+        }
+      });
     }
   }
-
-  static Cluster getCluster(InetSocketAddress contactPoint) {
```
Comment (codefromthecrypt) [inline zipkin-storage/cassandra-v1/src/test/java/zipkin2/storage/cassandra/v1/CassandraStorageExtension.java:154]: might not be obvious, but these settings repeated in the default session factory, except partially. moving it up allows better control over what we configure and also things like the same logging config to work in test sessions as the main sessions

File: zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/DefaultSessionFactory.java [hunk 2]
```diff
@@ -45,27 +47,23 @@
...
       if (cassandra.ensureSchema()) {
-        session = cluster.connect();
         Schema.ensureExists(keyspace, cassandra.searchEnabled(), session);
-        session.execute("USE " + keyspace);
       } else {
         LOG.debug("Skipping schema check on keyspace {} as ensureSchema was false", keyspace);
-        session = cluster.connect(keyspace);
       }
 
+      session.execute("USE " + keyspace);
```
Comment (anuraaga) [inline zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/DefaultSessionFactory.java:61]: Assuming connect / use is equivalent to `cluster.connect(keyspace)` from the old else branch

File: zipkin-storage/cassandra/src/test/java/zipkin2/storage/cassandra/CassandraStorageExtension.java [hunk 3]
```diff
@@ -65,42 +57,49 @@ Session session() {
...
 
-  CassandraStorage.Builder computeStorageBuilder() {
-    InetSocketAddress contactPoint = contactPoint();
+  CassandraStorage.Builder newStorageBuilder(TestInfo testInfo) {
     return CassandraStorage.newBuilder()
-        .contactPoints(contactPoint.getHostString() + ":" + contactPoint.getPort())
-        .ensureSchema(true)
-        .keyspace("test_cassandra3");
+      .contactPoints(contactPoint())
+      .maxConnections(1)
```
Comment (codefromthecrypt) [inline zipkin-storage/cassandra/src/test/java/zipkin2/storage/cassandra/CassandraStorageExtension.java:87]: hostdistance local 1 is exactly the same as this in impl.

**Review by anuraaga — APPROVED** [review]

File: zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/DefaultSessionFactory.java (hunk 2 above)
Comment (codefromthecrypt) [reply zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/DefaultSessionFactory.java:61]: yeah I looked into the code. basically cluster.connect(keyspace) is a composition of connect and use

**Comment by codefromthecrypt** [comment]
thanks for having a look @anuraaga!

**Pull request merged**